### ISSUE TITLE
"may be used uninitialized" warning in wallet/rpcwallet.cpp

### ISSUE THREAD

**Issue by kristapsk** [bitcoin/bitcoin#20381]
Have this warning while building current master (0bd4929cd00e91bf4137f43b097cd62f72f503e4) on one test machine.
```
wallet/rpcwallet.cpp: In function 'rescanblockchain()::<lambda(const RPCHelpMan&, const JSONRPCRequest&)>':
wallet/rpcwallet.cpp:3596:19: warning: '*((void*)& stop_height +4)' may be used uninitialized in this function [-Wmaybe-uninitialized]
 3596 |     Optional<int> stop_height;
      |                   ^~~~~~~~~~~
```
```
$ gcc --version
gcc (Gentoo 9.2.0-r2 p3) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ uname -a
Linux red-box 4.19.72-gentoo #2 SMP Sun Sep 22 03:00:01 EEST 2019 x86_64 Intel(R) Pentium(R) CPU N3710 @ 1.60GHz GenuineIntel GNU/Linux
```

**Comment by ryanofsky** [comment]
This can be fixed using a workaround like #18052, but the warning should be harmless, see #18198

**Comment by luke-jr** [comment]
Is there no way to put the workaround in the constructor?
